manga_auto_translator/translation.py

```diff
@@ -0,0 +1,38 @@
+from typing import Sequence
+from abc import ABC, abstractmethod
+from enum import Enum
+from manga_auto_translator.data_structure import Scan
+
+
+class TranslationStrategy(ABC):
+    @abstractmethod
+    def run(self, scans: Sequence[Scan], lang_from: str, lang_to: str) -> None:
+        raise NotImplemented()
+
+
+class GoogleTraductionTranslation(TranslationStrategy):
+    def __init__(self) -> None:
+        print('Loading GoogleTraductionTranslation...')
+        
+    def run(self, scans: Sequence[Scan], lang_from: str, lang_to: str) -> None:
+        from deep_translator import GoogleTranslator

```

Comment:
I am not a huge fan of imports in method scope, for testing purpose it may be better to pass it as argument.

Reply:
Maybe you have a reason to do like this, and I'm interested to know. It's not the first time I see this case, and some of greatest libraries used in ML are using this trick.

Reply:
I don't know about the google translator library, but MangaOCR takes literally forever to load. If someone uses another strategy, they would still need to wait for the library to load, even though they don't even use it. 

Importing from the function itself means that you only import it once, the first time you use a strategy that needs the library. 
So it's mostly to prevent loading unecessary modules.

Reply:
So if the loading time for the `GoogleTranslator` library is fast enough, the import should probably be placed with the other imports at the top of the file.

Reply:
Maybe we could create an anotation?
Something like:
```py
@module_argument(from='deep_translator' import='GoogleTranslator')
def run(self, scans: Sequence[Scan], lang_from: str, lang_to: str, GoogleTranslator: Module) -> None:
    ...
```

`Module` est probablement pas le bon type, mais l'idée est là.

Reply:
GoogleTranslator does not take long to import. So I've gone with the import at the top of the file.

For the decorator, we could do it, but for the translation in particular none of the other classes for the other api take long to import, so I do not see any benefit with the decorator.